utils/Cargo.toml

```diff
@@ -30,6 +30,8 @@ rand = { version = "0.8", optional = true }
 rand_core = { version = "0.6", optional = true }
 serde = "1.0"
 thiserror = { version = "1.0", optional = true }
+tokio = { version = "1.37.0", optional = true }
+wasm-bindgen-futures = { version = "0.4", optional = true }
```

Comment:
would it be possible to do something along the lines of:
```toml
[target.'cfg(target_family = "wasm")'.dependencies]
wasm-binggen-futures = { version = "0.4", optional = true }

[target.'cfg(not(target_family = "wasm"))'.dependencies]
tokio = { version = "1.37.0", optional = true }
```
as those dependencies are unnecessary otherwise. The exact syntax to facilitate this is not quite known to me though.

Reply:
Yeah, and then I'd use the `wasm-bindgen-futures` or `tokio` features in the rest of the code to select what gets enabled.

Reply:
Including the dependencies based on the target works fine 👍 I added it in db345de.

Reply:
@jsdanielh I don't think that using features instead of the configured target makes sense.

Generally, we want the codebase to be generic (i.e. platform-independent) as much as possible. This means that we want to keep the platform-specific code to a minimum and ideally keep it local as well. By using the target to select the correct implementation, we are encapsulating the platform-specific behaviour and contain it to this crate. The rest of the codebase can simply use the crate without having to worry about selecting the right features for the right target, keeping it generic. Using features would propagate the platform dependency to every consumer of this crate which is not what we want. The only advantage of using features would be that we get more control over which implementation is selected. However, there is not reasonable use case where we would want the tokio implementation for a wasm target or vice versa, so it would just result in unnecessary configuration overhead and more places to get wrong.